index.js

```diff
@@ -273,14 +387,24 @@ async function run() {
     const cluster = core.getInput('cluster', { required: false });
     const waitForService = core.getInput('wait-for-service-stability', { required: false });
     let waitForMinutes = parseInt(core.getInput('wait-for-minutes', { required: false })) || 30;
+
     if (waitForMinutes > MAX_WAIT_MINUTES) {
       waitForMinutes = MAX_WAIT_MINUTES;
     }
 
     const forceNewDeployInput = core.getInput('force-new-deployment', { required: false }) || 'false';
     const forceNewDeployment = forceNewDeployInput.toLowerCase() === 'true';
     const desiredCount = parseInt((core.getInput('desired-count', {required: false})));
+    const capacityProviderStrategy = core.getInput('capacity-provider-strategy',{required: false}) || [{base: 1, capacityProvider: 'FARGATE', weight: 1}];
 
+    if(capacityProviderStrategy){
+      core.debug('Using capacity provider ommits the launchtype option')
+      launchType = undefined;
+    }
+    else{
+      core.debug('Using launch type ommits the capacity provider option')
+      launchType = 'FARGATE' ;
+    }
```

Comment:
There seems be a logical error above. I will let you re-read the code and identify the error before pointing it out :-D
hint: It's related to the default value being set.

Reply:
Should i just delete the value that its being equal to ? Since launch type will be defined from the start by the user ?

Reply:
I meant couple of lines above but let me put it out. The default value for the `capacityProviderStrategy` is set at line no. 398 after which you are checking `if(capacityProviderStrategy){` which will always result in `true` and override the launchType value if provided by the customer.